operators/src/processing/vector_join/mod.rs

```diff
@@ -0,0 +1,191 @@
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+
+use geoengine_datatypes::collections::VectorDataType;
+
+use crate::engine::{
+    ExecutionContext, InitializedOperator, InitializedOperatorImpl, InitializedVectorOperator,
+    Operator, TypedVectorQueryProcessor, VectorOperator, VectorQueryProcessor,
+    VectorResultDescriptor,
+};
+use crate::error;
+use crate::util::Result;
+
+use self::equi_data_join::EquiLeftJoinProcessor;
+
+mod equi_data_join;
+
+/// The vector join operator requires two inputs and the join type.
+pub type VectorJoin = Operator<VectorJoinParams>;
+
+/// A set of parameters for the `VectorJoin`
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct VectorJoinParams {
+    #[serde(flatten)]
+    join_type: VectorJoinType,
+}
+
+/// Define the type of join
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(tag = "type")]
+pub enum VectorJoinType {
+    /// A left equi join between a `GeoFeatureCollection` and a `DataCollection`
+    EquiLeft {
```

Comment:
If we allow other types of joins in the future the type should maybe reflect more precisely what it does. Maybe call it `EquiLeftGeoToData` because in the future we might also support Data to Geo or Geo to Geo or Data to Data joins

Reply:
becf1fd